springwolf-plugins/springwolf-spring-cloud-rabbit-binder-plugin/src/main/java/io/github/stavshamir/springwolf/asyncapi/scanners/channels/CloudStreamRabbitBinderChannelsScanner.java

```diff
@@ -0,0 +1,198 @@
+import org.springframework.cloud.stream.config.BindingServiceProperties;
+import org.springframework.context.EmbeddedValueResolverAware;
+import org.springframework.context.annotation.Bean;
+import org.springframework.messaging.handler.annotation.Payload;
+import org.springframework.stereotype.Service;
+import org.springframework.util.StringValueResolver;
+import sun.reflect.generics.reflectiveObjects.ParameterizedTypeImpl;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.lang.reflect.Type;
+import java.util.*;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.Collectors.toSet;
+
+@Slf4j
+@Service
+@EnableConfigurationProperties({RabbitBinderConfigurationProperties.class, RabbitExtendedBindingProperties.class, BindingServiceProperties.class})
+public class CloudStreamRabbitBinderChannelsScanner implements ChannelsScanner, EmbeddedValueResolverAware {
+
+    private StringValueResolver resolver;
+    private final Map<String, Binding> bindingsMap;
+    private RabbitExtendedBindingProperties rabbitExtendedBindingProperties;
+    private BindingServiceProperties bindingServiceProperties;
+    private SchemasService schemasService;
+    private AsyncApiDocket docket;
+
+    public CloudStreamRabbitBinderChannelsScanner(AsyncApiDocket docket, RabbitExtendedBindingProperties rabbitExtendedBindingProperties, BindingServiceProperties bindingServiceProperties, SchemasService schemasService) {
+        this.docket = docket;
+        this.rabbitExtendedBindingProperties = rabbitExtendedBindingProperties;
+        this.bindingServiceProperties = bindingServiceProperties;
+        this.bindingsMap = makeBindingMap(rabbitExtendedBindingProperties, bindingServiceProperties);
+        this.schemasService = schemasService;
+    }
+
+    private Map<String, Binding> makeBindingMap(RabbitExtendedBindingProperties rabbitExtendedBindingProperties, BindingServiceProperties bindingServiceProperties) {
+        Map<String, StreamRabbitBindingInfo> bindingInfoMap = new HashMap<>();
+        rabbitExtendedBindingProperties.getBindings().entrySet().stream().forEach(entry -> {
+            StreamRabbitBindingInfo binding = new StreamRabbitBindingInfo();
+            binding.routingKey = entry.getValue() != null && entry.getValue().getConsumer() != null ? entry.getValue().getConsumer().getBindingRoutingKey() : null;
+            bindingInfoMap.put(entry.getKey(), binding);
+        });
+        bindingServiceProperties.getBindings().entrySet().stream().forEach(entry -> {
+            StreamRabbitBindingInfo binding = bindingInfoMap.containsKey(entry.getKey()) ? bindingInfoMap.get(entry.getKey()) : new StreamRabbitBindingInfo();
+            binding.exchange = entry.getValue().getDestination();
```

Comment:
Can `getValue()` return `null` here?

Reply:
I've never seen it.. but better be null safe just in case.